lib/std/mem.zig

```diff
@@ -150,9 +220,21 @@ pub const Allocator = struct {
         /// null means naturally aligned
         comptime alignment: ?u29,
         n: usize,
+    ) Error![]align(alignment orelse @alignOf(T)) T {
+        return self.alignedAlloc2(T, alignment, n, .exact);
+    }
+
+    const Exact = enum {exact,atLeast};
+    pub fn alignedAlloc2(
```

Comment:
I propose to rename this to `allocAdvanced` and deprecate `alignedAlloc`. Same idea with `alignedRealloc2`